helm/harbor-scanner-trivy/templates/statefulset.yaml

```diff
@@ -127,9 +126,11 @@ spec:
 {{ toYaml .Values.resources | indent 12 }}
           {{- end }}
       volumes:
-        {{- if .Values.persistence.enabled }}
         - name: tmp-data
           emptyDir: {}
+        {{- if not .Values.persistence.enabled }}
```

Comment:
This might be confusing. Shouldn't we disable readOnlyRootFilesystem if someone really doesn't like volumes?

I'd say by default we encourage using volumes for saving tmp data and scan reports.
If persistence is disabled, it means that someone really want to write to the containers file system, e.g. for the sake of testing

Reply:
I agree you should only disable persistence when testing.

Although, I think it makes sense to still use an emptyDir volume when having persistence disabled. This seems to be the most common way other Helm charts do it as well. Not sure what the exact reasoning behind it is, but I think you should prevent writing to the filesystem of the image itself. Also the data with an emptyDir will persist between container crashes, which could be useful (even if it's just for testing).

Reply:
The default deployment spec & helm values do not allow writing to container file system. We already have secure defaults.

For the sake of consistency I'd guard creation of both `tmp-data` and `data` volumes behind `.Values.persisntece.enabled` boolean value check.

IMO, Preserving emptyDir volume bring little value as most of the data is stored in Redis cluster. The data & tmp-data volumes store volatile data which are not needed anymore after container restarts.

Reply:
The data emptyDir volume should only be created when persistence is disabled, as it will be created by volumeClaimTemplates otherwise. This differs from tmp-data, which is never persistent.

For /tmp we either need to always use an emptyDir volume (whether persistence is enabled or not), or only create this when persistence is enabled and let the user disable the readOnlyRootFilesystem securityContext when they decide to disable persistence. I would prefer to always create an emptyDir volume, but am fine either way. If you prefer the latter I'll update the merge request with this.